core/tests/protractor_utils/forms.js

```diff
@@ -624,15 +624,19 @@ var CodeMirrorChecker = function(elem) {
    *  - 'checked': true or false, whether the line has been checked
    */
   var _compareTextFromLine = function(
-      currentLineNumber, scrollTo, compareDict) {
+      currentLineNumber, scrollTo, Pane, compareDict) {
+    /* eslint-disable */
     browser.executeScript(
-      "$('.CodeMirror-vscrollbar').first().scrollTop(" + String(scrollTo) +
+      "$('.CodeMirror-vscrollbar')." + Pane + ".scrollTop(" + String(scrollTo) +
       ');');
+    /* eslint-enable */
     elem.getText().then(function(text) {
       // The 'text' arg is a string 2n lines long representing n lines of text
       // codemirror has loaded. The (2i)th line contains a line number and the
       // (2i+1)th line contains the text on that line.
       var textArray = text.split('\n');
+      // Adding an empty line at the last.
+      textArray[textArray.length] = '';
```

Comment:
Try adding reasons for some "not obvious" code in the comment.

Reply:
I've added PTAL!

Reply:
`TextArray`-->`textArray`. Also, I don't think this explanation is sufficient. It does not answer why we need this. Try to explain that as well.

Reply:
@NishealJ, you missed the latter part of the comment: 

>Also, I don't think this explanation is sufficient. It does not answer why we need this. Try to explain that as well.

Reply:
Done Added now

Reply:
Hmm, @NishealJ, you are not actually adding an empty line, right? An empty line would be `'\n'`.  You're adding an empty string (or char).

Reply:
Also, I think this can be improved further:

```
// in codemirror5's getText() method. Therefore, we need to add an empty string at the end
// of the textArray to match with compareDict.
```
@DubeySandeep, thoughts?

Reply:
Hi @DubeySandeep & @apb7,
Array matches with dict i.e textArray and compareDict.
To add an empty line in compare dict we would have used a '\n' but as textArray is an array which is already split by '\n' we can just add '' to match with compare dict.
The below screenshot will help you understand the changes in getting a text of empty line from a codemirror5's Pane, you can use console.log(TextArray) to see the changes
<img width="699" alt="Screenshot 2019-06-28 at 12 24 37 PM" src="https://user-images.githubusercontent.com/41017303/60332535-b8282800-999f-11e9-9ea9-2b1c19bd3754.png">

Reply:
The suggested changed looks good to me updated Thanks!